File: mobile/lib/features/auth/data/models/user_summary.dart
class UserSummary {
  const UserSummary({
    required this.id,
    required this.fullName,
    required this.email,
    required this.currency,
    required this.salaryDay,
  });

  factory UserSummary.fromJson(Map<String, dynamic> json) {
    return UserSummary(
      id: json['id'] as String,
      fullName: json['fullName'] as String,
      email: json['email'] as String,
      currency: json['currency'] as String,
      salaryDay: json['salaryDay'] as int,
    );
  }

  final String id;
  final String fullName;
  final String email;
  final String currency;
  final int salaryDay;
}

File: backend/src/main/java/com/afterpay/identity/api/dto/UserSummary.java
package com.afterpay.identity.api.dto;

import java.math.BigDecimal;
import java.util.UUID;

import com.afterpay.identity.domain.User;

public record UserSummary(
    UUID id,
    String fullName,
    String email,
    BigDecimal monthlySalary,
    String currency,
    int salaryDay
) {
    public static UserSummary from(User user) {
        return new UserSummary(
            user.getId(),
            user.getFullName(),
            user.getEmail(),
            user.getMonthlySalary(),
            user.getCurrency(),
            user.getSalaryDay()
        );
    }
}
